inst/include/plant/models/ff16_environment.h

```diff
@@ -55,9 +54,23 @@ class FF16_Environment : public Environment {
     canopy.r_init_interpolators(state);
   }
 
-  virtual void compute_rates() {
+  // currently only used by FF16w but present in both FF16 and K93
+  // avoids uninitialised variables/splines because vars.state_size = 0
+  // perhaps not completely user safe, but I think diffcult to break from R API
+  virtual void compute_rates(std::vector<std::vector<double>> const& resource_depletion) {
+    // treat each soil layer as a separate resource pool
     for (size_t i = 0; i < vars.state_size; i++) {
-      vars.set_rate(i, extrinsic_driver_evaluate("rainfall", time) / (i+1));
+      double water_infiltration = extrinsic_drivers["rainfall"].eval(time) / (i+1);
```

Comment:
I think we can remove the `/ (i+1)` term here.

Reply:
Agree - it's a leftover test of soil depths, but will be replaced with drainage in time.